staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -649,7 +649,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 			return nil, nil, fmt.Errorf("unknown action verb for discovery: %s", action.Verb)
 		}
 
-		routes := []*restful.RouteBuilder{}
```

Comment:
I don't know if I can comment this, but I'd like to ask out of personal curiosity.
In term of flexibility for future logic changes, `routes` look fine.
Is it a bad idea to declare `routes` outside the for-loop?

```go
var routes []*restful.RouteBuilder
for _, action := range actions {
    ...
}
for _, route := range routes {
	ws.Route(route)
}
```

Reply:
Of course you may comment.

If in the future one desires to add a new route, they may change it again into a loop. For something this small, I personally do not see a point in leaving more complex code for extensibility's sake.

Reply:
That's right. This change is relatively simple, and if it needs to be changed in the future, it can be modified then.
I understood that it is better to avoid unnecessary complexity in the current state.
Thank you for your clear explanation!